dds_cli/__main__.py

```diff
@@ -1002,3 +946,128 @@ def abort(click_ctx, username, project):
+# USER MANAGEMENT ############################################### USER MANAGEMENT #
+###################################################################################
+@dds_main.group(no_args_is_help=True)
+@click.pass_obj
+def user(click_ctx):
+    """Manage user accounts, including your own."""
+    pass
+
+
+## Invite new users to the system
+@user.command()
+@click.argument(
+    "email",
+    nargs=1,
+    type=str,
+    required=True,
+    callback=dds_cli.utils.email_validator,
+)
+@click.option(
+    "--username",
+    "-u",
+    required=False,
+    type=str,
+    help="Your Data Delivery System username.",
+)
+@click.option(
+    "--role",
+    "-r",
+    required=True,
+    type=click.Choice(
+        choices=["Super Admin", "Unit Admin", "Unit Personnel", "Project Owner", "Researcher"],
+        case_sensitive=False,
+    ),
+    help="Type of account.",
+)
+@click.option(
+    "--project",
+    "-p",
+    required=False,
+    type=str,
+    help="Existing Project you want the user to be associated to.",
+)
+@click.pass_obj
+def add(click_ctx, username, email, role, project):
+    """
+    Add a user to the DDS system or hosted projects.
+
+    Specify an user's email and role to associate it with projects.
+    If the user doesn't exist in the system yet, an invitation email
+    will be sent automatically to that person.
+
+    """
+    try:
+        with dds_cli.account_manager.AccountManager(
+            username=username, no_prompt=click_ctx.get("NO_PROMPT", False)
+        ) as inviter:
+            inviter.add_user(email=email, role=role, project=project)
+            if project:
+                LOG.info(
+                    "Any user shown as invited would need to be added to the project "
+                    "once the user has accepted the invitation and created an account in the system."
+                )
+    except (
+        dds_cli.exceptions.AuthenticationError,
+        dds_cli.exceptions.ApiResponseError,
+        dds_cli.exceptions.ApiRequestError,
+        dds_cli.exceptions.DDSCLIException,
+    ) as err:
+        LOG.error(err)
+        sys.exit(1)
+
+
+## Delete users from the system
+@user.command(no_args_is_help=True)
+@click.argument(
+    "email",
+    nargs=1,
+    type=str,
+    required=False,
+    callback=dds_cli.utils.email_validator,
+)
+@click.option(
+    "--self",
+    required=False,
+    is_flag=True,
+    help="Decommission your own user account",
+)
+@click.pass_obj
+def delete(click_ctx, email, self):
+    """
+    Delete user accounts from the Data Delivery System.
+
+    To request the removal of your own account, use the `--self` flag without any arguments.
+    An e-mail will be sent to you asking to confirm the deletion.
+
+    If you have sufficient administrative privileges, you may also delete the accounts of other users.
+    Specify the e-mail address as argument to the main command to initiate the removal process.
+    """
+    if email:
```

Comment:
Add a check to make sure that either `email` or `self` is used

Reply:
If neither is specified, that would trigger the help message, since the command reads `@user.command(name="delete", no_args_is_help=True)`.